src/Dialect/ONNX/MLIRDialectBuilder.cpp

```diff
@@ -392,6 +392,12 @@ memref::CastOp MemRefBuilder::cast(Value input, MemRefType outputType) const {
 }
 
 Value MemRefBuilder::dim(Value val, int64_t index) const {
+  assert((val.getType().isa<MemRefType>() ||
+             val.getType().isa<UnrankedMemRefType>()) &&
```

Comment:
Can we get dim from an UnrankMemRefType?

Reply:
Yes, according to https://mlir.llvm.org/docs/Dialects/MemRef/#memrefdim-mlirmemrefdimop the source operand can have type "unranked.memref of any type values or memref of any type values"